imagescan/controller.go

```diff
@@ -151,15 +151,19 @@ func (s *Controller) handleDelta(event kube.Event, o kube.Object) {
 func (s *Controller) scheduleScans(ctx context.Context) (rerr error) {
 	s.syncFromRemoteState(ctx)
 
+	images := s.delta.getImages()
 	if s.fullSnapshotSent {
-		s.sendImagesResourcesChanges(ctx)
+		// Filter only images that have owner changes.
+		images = lo.Filter(images, func(img *image, _ int) bool {
+			return !img.ownerChanges.empty()
+		})
+		s.clearOwnerState()
 	} else {
-		// Send initial full images current resources state once.
-		if err := s.sendFullSnapshotImageResources(ctx); err != nil {
-			s.log.Errorf("sending initial full images resources changes: %v", err)
-		} else {
-			s.fullSnapshotSent = true
-		}
+		s.fullSnapshotSent = true
+	}
+
+	if err := s.updateImageStatuses(ctx, images, s.fullSnapshotSent); err != nil {
```

Comment:
lets move all this logic
```
images := s.delta.getImages()
	if s.fullSnapshotSent {
		// Filter only images that have owner changes.
		images = lo.Filter(images, func(img *image, _ int) bool {
			return !img.ownerChanges.empty()
		})
		s.clearOwnerState()
	} else {
		s.fullSnapshotSent = true
	}
```
	
	
to updateImageStatuses as it's needed only for that method to work.


* Also s.fullSnapshotSent should be sent to true only after successfull send.